src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -383,8 +383,27 @@ namespace winrt::TerminalApp::implementation
                 {
                     if (const auto scheme = _settings.GlobalSettings().ColorSchemes().TryLookup(realArgs.SchemeName()))
                     {
+                        // Start by getting the current settings of the control
                         auto controlSettings = activeControl.Settings().as<TerminalSettings>();
-                        controlSettings.ApplyColorScheme(scheme);
+                        auto parentSettings = controlSettings;
+                        // Those are the _runtime_ settings however. What we
+                        // need to do is:
+                        //
+                        //   1. Blow away any colors set in the runtime settings.
+                        //   2. Apply the color scheme to the parent settings.
+                        //
+                        // 1 is important to make sure that the effects of
+                        // something like `colortool` are cleared when setting
+                        // the scheme.
+                        if (controlSettings.GetParent() != nullptr)
+                        {
+                            parentSettings = controlSettings.GetParent();
+                        }
+
+                        // ApplyColorScheme(nullptr) will clear the old color scheme.
+                        controlSettings.ApplyColorScheme(nullptr);

```

Comment:
Have a question about this - why do we clear the runtime colors? I understood the 'control-decided' settings to only be mutable by the control. 

Also, wouldn't this mean that the 'PreviewColorScheme' appearance may not match up to the 'SetColorScheme' appearance? Because it seems that we _don't_ clear out the runtime colors when previewing the new scheme, but we do when we set the new scheme (so if there were some colors in the runtime settings, they will show up in the preview but will disappear after setting the new scheme)

Reply:
Yea, this is a rough one. It probably should only be mutable by the control, but if it _is_, then the SetColorScheme action would do _nothing at all_ when the user's already run `colortool`, which is not really what we want, right? So the only option here is to clear out the runtime settings.

Weirdly enough, this seemed to work fine when doing the colortool -> preview a scheme. I agree that it seems like it shouldn't. I could dig into it and investigate further if need be